docs/apis/subsystems/xapi/index.md

```diff
@@ -0,0 +1,350 @@
+The xAPI specification can be consulted in the xAPI specs page: https://github.com/adlnet/xAPI-Spec.
+
+## Moodle xAPI functionalities
+
+The final objective of Moodle xAPI library is NOT to implement a full LRS inside Moodle but to provide an easy way to handle xAPI statements within any plugin that needs it.
+
+For now, Moodle supports the main xAPI request which is called "**statement**". A statement must be seen as a tracking message that can be used by any plugin to store user activity directly to the Moodle without programming any additional web service.
+
+The current library implements:
+
+- A new webservice called **core_xapi_statement_post** to process xAPI statements and generate standard Moodle events.
+- A class **\core_xapi\handler** that any plugin can extend in order to user xAPI and generate specific plugin contexts. This way, any plugin could use xAPI without implementing any new web service.
+- A class **\core_xapi\iri** to easily translate random information into valid IRI values (needed for xAPI objects and verbs)
+- A class **\core_xapi\local\statement** to create and extract information from statement data.
+- A bunch of predefined classes in **\core_xapi\local\statement\item_XXX** to generate xAPI statement items from the plugins (example provided in [Generating statements in PHP](#generating-statements-in-php)).
+
+## Statement processing
+
+The `core_xapi_statement_post` webservice receives a JSON encoded statement (could be an array or a single one) and a component frankenstyle name.
+
+The processing sequence is:
+
+1. The core xAPI library checks if the specified component has a xAPI handling class implementation. Otherwise it returns an error.
+1. xAPI checks the statement structure, if any check fails, it returns an error. The main validations for now are:
+   1. Check that only supported xAPI attributes are used. If the statement array has some format error all statements will be rejected (see [Statement data validation](#statement-data-validation) for more information).
+   1. Check all users and groups are created in the LMS.
+   1. Check the current user ($USER) is an actor of the statement (actors can be a single agent or a group).
+1. For every statement:
+   1. xAPI asks the component handler to convert the statement into a standard event (function `statement_to_event()`). In case the plugin could not convert some statement into an event, this statement will be marked as "not processed" and will continue to the next statement.
+   1. xAPI triggers the generated events.
+1. Once all statements are processed:
+   1. If ALL statements are marked as "not processed" it returns an error.
+   1. If some statement could be processed, returns an array containing "true" if the statement is processed and "false" otherwise.
+
+:::note
+
+The function `statement_to_event()` is responsible for:
+
+- Grant that the specified user has permission to execute that statement.
```

Comment:
"Granting"? Although "Asserting" would read clearer

Reply:
Following lines should be "Providing..." & "Processing..."

Reply:
Replaced Grant to Asserting; and also Providing and Processing